### PR TITLE
Use docker networks and aliases instead links in tests

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: tests/docker_cluster.py [hunk 1]
```diff
@@ -178,6 +182,18 @@ def _create_host_mount_dirs(self):
...
+        self.client.networks.create(self._network_name)
+
+    def _get_network(self):
+        return self.client.networks.get(self._network_name)
+
+    def _remove_network(self):
+        try:
+            self._get_network().remove()
+        except NotFound:
+            pass
```
Comment (kokosing) [inline tests/docker_cluster.py]: Why is so? Maybe a warning?

File: tests/docker_cluster.py [hunk 2]
```diff
@@ -206,8 +223,14 @@ def _create_container(self, image, container_name, hostname, cmd, **kwargs):
                                volumes={master_mount_dir: {'bind': self.mount_dir, 'mode': 'rw'}},
                                command=cmd,
                                mem_limit='2g',
+                               network=None,
+#                               network=self._network_name,
```
Comment (kokosing) [inline tests/docker_cluster.py]: ?

File: tests/docker_cluster.py [hunk 3]
```diff
@@ -206,8 +223,14 @@ def _create_container(self, image, container_name, hostname, cmd, **kwargs):
...
                                **kwargs)
 
+        self._get_network().connect(
+            container_name,
+            aliases=[hostname.split('-')[0]])
```
Comment (kokosing) [inline tests/docker_cluster.py]: why do you use `split` here `-`? How `hostname` look like?

File: tests/docker_cluster.py [hunk 4]
```diff
@@ -190,8 +206,9 @@ def _create_and_start_containers(self, master_image, slave_image=None, cmd=None,
             self.master,
             hostname=self.internal_master,
             cmd=cmd,
-            links=zip(self.slaves, self.slaves),
             **kwargs)
+
+        self._get_network().connect(self.master)
```
Comment (kokosing) [inline tests/docker_cluster.py]: master should be already connected, see line 204 where you call `_create_container`

File: tests/product/base_product_case.py [hunk 5]
```diff
@@ -85,10 +85,7 @@ class BaseProductTestCase(BaseTestCase):
 query.max-memory-per-node=512MB
 query.max-memory=50GB\n"""
 
-    # The two strings below (down_node_connection_string and status_down_node_string) aggregate
-    # all possible error messages one might encounter when trying to perform an action when a
-    # node is not accessible. The variety in error messages comes from differences in the OS.
-    down_node_connection_string = r'(\nWarning: (\[%(host)s\] )?Low level ' \
+    network_unreachable = r'(\nWarning: (\[%(host)s\] )?Low level ' \
```
Comment (kokosing) [inline tests/product/base_product_case.py]: Is this one is still valid once you ported tests to use network? I thought only one of these two errors should be expected.

**Comment by kokosing** [comment]
Also make sure travis is passing. Other than that you did a great job!

**Comment by kokosing** [comment]
Fixes #348 and #351

File: tests/docker_cluster.py (hunk 1 above)
Comment (losipiuk) [reply tests/docker_cluster.py]: I think it is not need. Will remove.

**Pull request closed without merging**